hypertrace-core-graphql-rx-utils/src/main/java/org/hypertrace/core/graphql/rx/NetworkSchedulerProvider.java

```diff
@@ -0,0 +1,46 @@
+package org.hypertrace.core.graphql.rx;
+
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import io.reactivex.rxjava3.core.Scheduler;
+import io.reactivex.rxjava3.schedulers.Schedulers;
+import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ThreadFactory;
+import javax.inject.Inject;
+import javax.inject.Provider;
+import org.hypertrace.core.graphql.spi.config.GraphQlServiceConfig;
+
+/**
+ * A scheduler using up to a configuration-based number of threads.
+ *
+ * <p>This differs from the default Scheduler.io implementation which uses an unbounded number of
```

Comment:
Where is Scheduler.io implementation present? 
Also, do we have any metrics around the default thread settings?

Reply:
http://reactivex.io/RxJava/3.x/javadoc/io/reactivex/rxjava3/schedulers/Schedulers.html#io--
What metrics are you looking for? The main concern with it is that it's unbounded.